src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/dto/MigrateRequest.kt

```diff
@@ -0,0 +1,55 @@
+package uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.dto
+
+import io.swagger.v3.oas.annotations.media.Schema
+import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.SYSTEM_USERNAME
+import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.jpa.NonAssociation
+
+@Schema(description = "Migrate Request")
+data class MigrateRequest(
+  @Schema(description = "Prisoner number to not associate", required = true, example = "A1234BC")
+  val firstPrisonerNumber: String,
+  @Schema(
+    description = "Reason why this prisoner should be kept apart from the other",
+    required = true,
+    example = "VICTIM",
+  )
+  val firstPrisonerReason: NonAssociationReason,
+  @Schema(description = "Prisoner number to not associate", required = true, example = "D5678EF")
+  val secondPrisonerNumber: String,
+  @Schema(
+    description = "Reason why this prisoner should be kept apart from the other",
+    required = true,
+    example = "PERPETRATOR",
+  )
+  val secondPrisonerReason: NonAssociationReason,
+
+  @Schema(description = "Type of restriction, e.g. don't locate in the same cell", required = true, example = "CELL")
+  val restrictionType: NonAssociationRestrictionType,
+
+  @Schema(
+    description = "Type of restriction, e.g. don't locate in the same cell",
```

Comment:
the description probably needs tweaking to something like `"Explanation of why prisoners are non-associated"`, given it's for the comment field :)